### ISSUE TITLE
False positive in Lint/UnusedMethodArgument when the argument is reasigned in a lambda

### ISSUE THREAD

**Issue by rrosenblum** [bbatsov/rubocop#4898]
`Lint/UnusedMethodArgument` registers an offense when an argument is being reassigned in a lambda before the argument is used in the method regardless of if the lambda is being called. I think this issue was introduced by #4826. I believe the issue is happening in `UnusedArgument#variable_used?`. This issue likely also exists in `Lint/UnusedBlockArgument`.

```ruby
def something=(value)
  convert = ->(str) { value = str.split(',') }
  @something = value.is_a?(String) ? convert.call(value) : value # `@something = value` will also cause the offense
end
```

## RuboCop version
```
$ rubocop -V
0.51.0 and master (using Parser 2.4.0.0, running on ruby 2.4.2 x86_64-darwin16)
```
